### PR TITLE
Alex Brandt Integrate ChatGPT

### PR DESCRIPTION
**Pull request by brandta-1**
# Description
image

## Related PRS (if any):
This PR was built off of [646](https://github.com/OneCommunityGlobal/HGNRest/pull/646)
To test this backend PR you need to checkout the #[2083](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2083) frontend PR.

### ```src/controllers/chatGPTController.js```
This controller takes in a front-end request containing the user's id and name. This data is used to retrieve their weekly summaries, then the controller retrieves the latest AI prompt, and sends this data to ChatGPT, and sends the AI response to the front end.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Create a ChatGPT API key (or use an existing one) and set it as the ```OPENAI_API_KEY``` environment variable in your ```.env``` file
image

## How to get a ChatGPT API key

1. Go to [OpenAI's API portal](https://auth0.openai.com/u/signup/identifier?state=hKFo2SBmV2YzamhMVkZ0NmtoTGFUdE14YWZMazNWNl85V1RQUKFur3VuaXZlcnNhbC1sb2dpbqN0aWTZIF9vVWdUdlI0TFd1bWRDWTJfMlJNREMxR0lsZW5QWWZJo2NpZNkgRFJpdnNubTJNdTQyVDNLT3BxZHR3QjNOWXZpSFl6d0Q)
2. Create an account (or login to an existing one) and navigate to API keys on the left side of the page:
image
3. Create a new secret key:
image
4. Name the key (optional) the click create secret key:
image
5. copy the key string and save it somewhere, then paste it into your ```.env``` file:
image

## Screenshots or videos of changes:
See files changed

### CODE REVIEW

**Review by Abi-Liu — COMMENTED** [review]
I'm getting an error when trying to run `npm install` on the backend. Do you know what may be causing this error?


Screenshot 2024-03-22 at 1 58 28 PM

File: src/.DS_Store
Comment (DiegoSalas27) [inline src/.DS_Store:1]: Please delete this file and add it to the .gitignore file.

**Review by DiegoSalas27 — CHANGES_REQUESTED** [review]

File: src/.DS_Store
Comment (brandta-1) [reply src/.DS_Store:1]: [added](https://github.com/OneCommunityGlobal/HGNRest/pull/809/commits/8dea04f1c1f05eafd1ed6f3083e95d4b8efbc32f)

**Comment by brandta-1** [comment]
> I'm getting an error when trying to run `npm install` on the backend. Do you know what may be causing this error?
> 
> Screenshot 2024-03-22 at 1 58 28 PM

It looks like this was an error with json format in package-lock, the error should be fixed now. If it persists, then try deleting ```package-lock.json```, thank you.

**Review by jennZhang93 — COMMENTED** [review]
Please see the [frontend review](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2083#pullrequestreview-1990697661)

**Comment by KaushikMreddy** [comment]
Requested Changes and  reviewed this PR at the front end. 
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2083#pullrequestreview-2047073862

Running the application was simple and generating the API key was easy, Thank you for adding the steps to create one.

**Pull request closed without merging**